crates/nu-command/src/platform/input/list.rs

```diff
@@ -0,0 +1,210 @@
+use dialoguer::MultiSelect;
+use dialoguer::{console::Term, theme::ColorfulTheme, Select};
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, IntoPipelineData, PipelineData, ShellError, Signature, SyntaxShape, Type,
+    Value,
+};
+
+enum InteractMode {
+    Single(Option<usize>),
+    Multi(Option<Vec<usize>>),
+}
+
+#[derive(Clone)]
+pub struct InputList;
+
+impl Command for InputList {
+    fn name(&self) -> &str {
+        "input list"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("input list")
+            .input_output_types(vec![(
+                Type::List(Box::new(Type::Any)),
+                Type::List(Box::new(Type::Any)),
+            )])
+            .optional("prompt", SyntaxShape::String, "the prompt to display")
+            .switch(
+                "multi",
+                "Use multiple results, you can press a to toggle all options on/off",
+                Some('m'),
+            )
+            .allow_variants_without_examples(true)
+            .category(Category::Platform)
+    }
+
+    fn usage(&self) -> &str {
+        "Interactive list selection. Abort with esc or q"
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["prompt", "ask", "menu"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let head = call.head;
+        let prompt: Option<String> = call.opt(engine_state, stack, 0)?;
+
+        let options: Vec<String> = match input {
```

Comment:
i do not quite understand the way you create `options` here :open_mouth: 

1. when i apply the following diff, it does not appear to change the behaviour of the `input list` command, so i'd say the `print`s are useless
```diff
diff --git a/crates/nu-command/src/platform/input/list.rs b/crates/nu-command/src/platform/input/list.rs
index c63f8303b..2075b748a 100644
--- a/crates/nu-command/src/platform/input/list.rs
+++ b/crates/nu-command/src/platform/input/list.rs
@@ -89,8 +89,6 @@ impl Command for InputList {
                         .iter()
                         .zip(record.1.iter())
                         .map(|(col, val)| {
-                            println!("col: {:?}", col);
-                            println!("val: {:?}", val);
                             if let Ok(val) = val.as_string() {
                                 Some(format!(" {}: {} |", &col, &val))
                             } else {
@@ -102,14 +100,12 @@ impl Command for InputList {
                 .collect(),
             PipelineData::Value(
                 Value::Record {
-                    cols,
+                    cols: _,
                     vals,
                     span: _,
                 },
                 _,
             ) => {
-                println!("cols: {:?}", cols);
-
                 vals.iter()
                     .map(|x| x.as_string().unwrap_or_else(|_| "RECORD".to_string()))
                     .collect()
```
2. in the first arm of `match` you have `options.push(...)` but not in the other arms
3. the `{col}: {val} |` format comes from `dialoguer`

> **Note**
> i might just be a noob with `dialoguer` here :laughing:

Reply:
> when i apply the following diff, it does not appear to change the behaviour of the input list command, so i'd say the prints are useless

i think they used the `println!()`s for debugging and were forgotten

Reply:
Yes see the discord thread this is the attempt to support records, this will be removed if I don't find how to handle Dates properly

Reply:
i see :+1: 

@melMass we can make this a DRAFT if it's not ready :relieved:

Reply:
Feel free to! Maintainers edits are enabled, but as said here and on discord I will either make it work today or remove it (table/record support)

Reply:
okey, i'll just make it a draft to avoid unwanted merge :relieved: 
you can mark it back as "ready for review" as soon as you fix or remove this :wink: